src/main/java/pl/zimi/zadanka/Zad5.java

```diff
@@ -0,0 +1,48 @@
+package pl.zimi.zadanka;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * source:
+ * https://leetcode.com/problems/climbing-stairs/
+ */
+
+public class Zad5 {
+
+    // rozwiązanie poprawne ale wolne
+    public static int climbStairsTreeStyle(int key, int currentValue, int carry) {
```

Comment:
generalnie nie mam nic do nazwy... ale nie bardzo wiem dlaczego TreeStyle... nie mówię że jest źle bo nie wiem czemu twierdzisz że to jest "tree style"

Reply:
dwie równoległe rekurencje badające wszystkie rezultaty dwóch operacji skojarzyły mi się z drzewem binarny, gdzie liczba to węzeł, a operacja dodania 1 lub 2 to ścieżka do węzła. Nie poprawiam teraz nazwy póki nie ustalimy, czy moje skojarzenie jest mylne

Reply:
zostałbym po prostu przy rekurencji: "RecurrenceStyle"
wywołania metod faktycznie układają się w drzewa (niekoniecznie binarne - jak lekko zmodyfikujesz problem)... ale to jest typowe dla rekurencji...

jak myślę o drzewach i używaniu tego pojęcia to myślę o drzewiastych strukturach danych... a takich tutaj brak... dlatego nie stosowałbym tej nazwy

lista, graf, drzewo, mapa to struktury danych nie używałbym nazw jeśli nie posiadamy odpowiedniej struktury

Reply:
ok dotarło